qiskit/circuit/add_control.py

```diff
@@ -121,12 +122,9 @@ def control(operation: Union[Gate, ControlledGate],
         qc.mcrz(operation.definition.data[0][0].params[0], q_control, q_target[0],
                 use_basis_gates=True)
     else:
-        if num_ctrl_qubits == 1:
-            basis_gates = ['x', 'z', 'y', 'u1', 'u3', 'cx']
-        else:
-            basis_gates = ['x', 'u1', 'u3', 'cx']
-
+        basis_gates = ['x', 'y', 'z', 'h', 'rx', 'ry', 'swap', 'ccx', 'u1', 'u3', 'cx']
```

Comment:
Is there a reason you didn't include `'rz'`?

Reply:
RZ and CRZ were causing an unwanted global phase. Although this shouldn't be an issue, this was causing some tests to fail. However, Unrolling without rz will give u1 gates instead. These gates gets controlled using a single mcu1 and does not result in a phase, nor require more gates to implement. 

Example: controlling RYY gate decomposition with and without RZ:
With RZ:
![ryy control_with_rz](https://user-images.githubusercontent.com/28988727/87885792-bc970700-ca20-11ea-8da3-8b304a0ee38c.png)


without RZ:
![ryy control_without_rz](https://user-images.githubusercontent.com/28988727/87885799-c4ef4200-ca20-11ea-87c8-2bca7351b794.png)

Reply:
We could unroll to to Rz but when controlling Rz we use cu1 instead of mcrz. This basically what the lines 58-66 do to get around the same problem. Or we could simply fix the test that is causing this issue but I'm not sure how.

Reply:
Now that Rz produces the correct matrix, the test is fixed. and now we can unroll to Rz.